secret_handlers.go

```diff
@@ -120,3 +120,38 @@ func SecretGet(c echo.Context) error {
 
 	return c.JSON(http.StatusOK, secret.ToSecretResponse())
 }
+
+func SecretEdit(c echo.Context) error {
+	secretDao, err := getSecretDao(c)
+	if err != nil {
+		return err
+	}
+
+	updateRequest := &m.SecretEditRequest{}
+	err = c.Bind(updateRequest)
+	if err != nil {
+		return util.NewErrBadRequest(err)
+	}
+
+	paramID, err := util.InterfaceToInt64(c.Param("id"))
+	if err != nil {
+		return err
+	}
+
+	secret, err := secretDao.GetById(&paramID)
+	if err != nil {
+		return err
+	}
+
+	err = secret.UpdateSecretFromRequest(updateRequest)
+	if err != nil {
+		return util.NewErrBadRequest(err)
+	}
+
+	err = secretDao.Update(secret)
+	if err != nil {
+		return util.NewErrBadRequest(err)
+	}
```

Comment:
just thinking 🤔 when provided id exists and `secret.UpdateSecretFromRequest()` doesn't return an err, what situation will lead to bad request err here? I think we can have here only `return err`

Reply:
yes 5xx is more relevant here, thanks!